bin/test/main.ml

```diff
@@ -100,6 +100,42 @@ let run_cram_tests ?root ppf temp_file pad tests t =
     ) tests;
   Block.pp_footer ppf ()
 
+let envs = Hashtbl.create 8
+
+let rec save_summary acc s =
+  let open Env in
+  match s with
+  | Env_value (summary, id, _) ->
+     save_summary (Ident.name id :: acc) summary
+  | Env_module (summary, id, _)
+    | Env_class (summary, id, _) ->
+     save_summary (Ident.unique_toplevel_name id :: acc) summary
```

Comment:
I suggest:

```
diff --git a/bin/test/main.ml b/bin/test/main.ml
index 597a117..e445166 100644
--- a/bin/test/main.ml
+++ b/bin/test/main.ml
@@ -108,11 +108,17 @@ let rec save_summary acc s =
   | Env_value (summary, id, _) ->
      save_summary (Ident.name id :: acc) summary
   | Env_module (summary, id, _)
-    | Env_class (summary, id, _) ->
-     save_summary (Ident.unique_toplevel_name id :: acc) summary
+    | Env_class (summary, id, _)
+    | Env_extension (summary, id, _) ->
+      let acc =
+        if Ident.binding_time id >= 1000
+        then Ident.unique_toplevel_name id :: acc
+        else acc
+      in
+      save_summary acc summary
   | Env_empty -> acc
   | Env_modtype (summary, _, _)
-    | Env_extension (summary, _, _)
     | Env_cltype (summary, _, _)
     | Env_open (summary, _)
     | Env_functor_arg (summary, _)
```

Identifiers with a binding_time between 1 and 1000 are handled differently (see `Predef` module).

Reply:
Thank you for your help :) 
But for some reason I still have some issues with modules, I get  and `Unbound module` error when reloading the objects on an existing environment. The name of the module is not in the summary of the environment, and it is not even in the module list (when calling `Env.fold_modules`).